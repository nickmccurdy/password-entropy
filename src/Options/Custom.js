import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import PropTypes from 'prop-types'
import { FormGroup, Input, Label } from 'reactstrap'

export default function Custom({ custom, onCustom }) {
  return (
    <FormGroup>
      <Label>
        <FontAwesomeIcon icon={faQuestionCircle} /> Possible Items
        <Input
          value={custom}
          onChange={onCustom}
          type="number"
          min="0"
          required
        />
      </Label>
    </FormGroup>
  )
}

Custom.propTypes = {
  custom: PropTypes.number.isRequired,
  onCustom: PropTypes.func.isRequired,
}
