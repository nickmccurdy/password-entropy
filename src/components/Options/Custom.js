import TextField from 'material-ui/TextField'
import Edit from 'material-ui-icons/Edit'
import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'

import { selector, setCustom } from '../../reducers/custom'

export function Custom({ possibleItems, setCustom }) {
  return (
    <TextField label={<div><Edit /> Possible Items</div>} value={possibleItems} onChange={setCustom} type="number" min="0" required />
  )
}

Custom.propTypes = {
  possibleItems: PropTypes.number.isRequired,
  setCustom: PropTypes.func.isRequired
}

export default connect(selector, { setCustom })(Custom)
