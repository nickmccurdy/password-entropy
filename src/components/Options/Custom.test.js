import { shallow } from 'enzyme'
import React from 'react'

import { Custom } from './Custom'

test('Custom', () => {
  const setCustom = jest.fn()
  const wrapper = shallow(<Custom possibleItems={1} setCustom={setCustom} />)
  wrapper.find('TextField').simulate('change')

  expect(wrapper).toMatchSnapshot()
  expect(setCustom).toHaveBeenCalled()
})
