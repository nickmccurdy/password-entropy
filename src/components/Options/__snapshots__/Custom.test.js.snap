// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom 1`] = `
<TextField
  disabled={false}
  floatingLabelFixed={false}
  floatingLabelText={
    <div>
      <ImageEdit />
       Possible Items
    </div>
  }
  fullWidth={false}
  min="0"
  multiLine={false}
  onChange={[Function]}
  required={true}
  rows={1}
  type="number"
  underlineShow={true}
  value={1}
/>
`;
